### ISSUE TITLE
IAM Policy Scanner Can Not Find Violations on Public Dataset

### ISSUE THREAD

**Issue by blueandgold** [forseti-security/forseti-security#3406]
In the test environment, this is a dataset that is configured with `allAuthenticatedUsers`, but the IAM policy scanner is unable to find this violation with the following rule, despite supporting it.  But this works on buckets with a similar IAM policy, for which violation is found.

```
IAM_TYPE_RESOURCE_MAP = {
   ResourceType.BILLING_ACCOUNT: BillingAccount,
   ResourceType.BUCKET: Bucket,
    ResourceType.DATASET: Dataset,
   ResourceType.FOLDER: Folder,
   ResourceType.ORGANIZATION: Organization,
   ResourceType.PROJECT: Project,
}
```

Example Rule:
```
  - name: Prevent public authenticated users from having access to buckets via IAM
    mode: blacklist
    resource:
      - type: dataset
        applies_to: self
        resource_ids:
          - '*'
    inherit_from_parents: true
    bindings:
      - role: '*'
        members:
          - allAuthenticatedUsers
```

**Comment by kevensen** [comment]
I am able to get a violation with `allAuthenticatedUsers`  given the role "roles/viewer" on a dataset.  However, when setting the role "roles/bigquery.dataViewer" I do not.

**Comment by kevensen** [comment]
I've located the root cause.  Adding `allAuthenticatedUsers` to the **roles/bigquery.dataViewer** role results in two members being assigned that role.

1. allAuthenticatedUsers
2. viewers of project: ....

So that results in two members of assigned to the binding

**Comment by kevensen** [comment]
@blueandgold It has to do with **projectViewer** not being in _member_types_.

The exception message on 189 is misleading because it is reporting the **members** in the binding tather than the invalid member type in **class IamPolicyBinding**.

So there are a couple issues.
1. Update the caught exception to log the message from the source of the exception.
2. Add **projectViewer** to the approved member types.

**Comment by kevensen** [comment]
@blueandgold In thinking about a solution here, I am curious if we even need the check for a valid member type.  Could we consider any member type returned by GCP valid?

Alternatively, we adjust the exception logic.  If a member has an invalid member type, we drop the member from the binding instead of the binding all together.

**Comment by blueandgold** [comment]
I double-checked the IAM policy for our test dataset with this violation, and it seems like there are other IAM policy members other than just `projectViewer`.  There are `projectEditor` and `projectOwner` as well.  Please see below.

I think that your proposal to only drop the unknown member from the binding (instead of all the binding all together), is the best change to make.

```
{
  "bindings": [
    {
      "members": [
        "projectEditor:inventoryscanner-henry"
      ],
      "role": "roles/bigquery.dataEditor"
    },
    {
      "members": [
        "projectOwner:inventoryscanner-henry",
        "user:henry@henrychang.mygbiz.com"
      ],
      "role": "roles/bigquery.dataOwner"
    },
    {
      "members": [
        "allAuthenticatedUsers",
        "projectViewer:inventoryscanner-henry",
        "user:goldspin@gmail.com"
      ],
      "role": "roles/bigquery.dataViewer"
    }
  ],
  "etag": "BwVktvfjU5M="
}
```

**Comment by kevensen** [comment]
I imagined there were others beyond "projectViewer".  I will adjust the exception logic to drop the member instead of the whole binding.

**Comment by blueandgold** [comment]
@kevensen Is this closed by your PR #3408?

**Comment by kevensen** [comment]
@blueandgold yes it is.
